hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -2296,6 +2546,35 @@ int SaraNcpClient::modemHardReset(bool powerOff) {
     return SYSTEM_ERROR_NONE;
 }
 
+int SaraNcpClient::modemEmergencyHardReset() {
+    if (ncpId() != PLATFORM_NCP_SARA_R510) {
+        return SYSTEM_ERROR_NONE;
+    }
+
+    LOG(TRACE, "Emergency hardware shutdown the modem");
+    const auto pwrState = modemPowerState();
+    // We can only reset the modem in the powered state
+    if (!pwrState) {
+        LOG(ERROR, "Modem is not powered on!");
+        return SYSTEM_ERROR_INVALID_STATE;
+    }
+
+    // Low held on power pin
+    HAL_GPIO_Write(UBPWR, 0);
+    HAL_Delay_Milliseconds(500);
+    // Low held on reset pin
+    HAL_GPIO_Write(UBRST, 0);
+    // Release power pin after 23s (23.5)
+    HAL_Delay_Milliseconds(23000);
```

Comment:
Because of the `HAL_Delay_Milliseconds(23000);` the reset pin is also held low for 23 sec. And in [this line](https://github.com/particle-iot/device-os/pull/2359/files#diff-a8e99f7ec884e049b049347a881b289b8edd5641af45e8d407cb97a78fdf7550R2571) it is held low for another 2 sec. From the data sheet, looks like 2 sec is just enough. And the RESET is held low _almost after_ the 23 sec of the low-PWR_ON. 
![image](https://user-images.githubusercontent.com/48732259/134192074-5a23e28f-8116-4e20-8309-70c6ffc55368.png)

Reply:
Got it, thanks for clarifying. Since you have tested it, I am totally ok.